### PR TITLE
Immersive colony start

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #
Closes #
Closes #

# Changes proposed in this pull request:
- Add data driven stories to abandoned supplycamps
- Add data driven stories to abandoned supplyships
- Add stories to abandoned colonies
- Server-driven colony creation (pre-validation before townhall opening)
- New UI for colony deletion/abandoning
- New UI for colony reactivation
- New UI for colony creation (includes entering name)
- New translation strings with more immersive messages for all of these things
- More relaxed placement of supplyship/camp taking the placeholders into account (will need some schematic adjustments possibly)
- Fix issues with supplycamp/ship placement handling

[ x ] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/core/client/gui/WindowSupplyStory.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        this.type = type;
+        this.stack = stack;
+        this.hand = hand;
+
+        registerButton(BUTTON_CANCEL, this::close);
+        registerButton(BUTTON_COLONY_SWITCH_STYLE, this::switchPack);
+        registerButton(BUTTON_PLACE, this::place);
+
+        final Random random = new Random(stack.getTag().getLong(TAG_RANDOM_KEY));
+        String story = "";
```
Comment (Nightenom) [inline src/main/java/com/minecolonies/core/client/gui/WindowSupplyStory.java]: this hurts + never use \n in component code -> use either list or component concatenation or text builder (or any combination)

File: src/main/java/com/minecolonies/core/client/gui/townhall/WindowTownHallDeleteAbandonColony.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    private void confirmDeleteColony()
+    {
+        Network.getNetwork().sendToServer(new ColonyDeleteOwnMessage());
+        Network.getNetwork().sendToServer(new GetColonyInfoMessage(pos));
+        close();
+    }
+
+    private void abandonColony()
+    {
+        this.findPaneOfTypeByID("dialog", Image.class).show();
```
Comment (Nightenom) [inline src/main/java/com/minecolonies/core/client/gui/townhall/WindowTownHallDeleteAbandonColony.java:83]: these methods are code smell imho, use view to wrap disabling/enabling of element groups, or use switch view (if applicable)

File: src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+/**
+ * Loads and listens to custom visitor data added
+ */
+public class ColonyStoryDataListener extends SimpleJsonResourceReloadListener
+{
+    /**
+     * Gson instance
+     */
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
```
Comment (Nightenom) [inline src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java:22]: _such a nice place for codecs xd_

File: src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * List of custom visitor data
+     */
+    public static List<String> abandonedColonyNames = new ArrayList<>();
+    public static List<String> abandonedColonyStories = new ArrayList<>();
+
+    public static List<String> supplyShipStories = new ArrayList<>();
+    public static List<String> supplyCampStories = new ArrayList<>();
```
Comment (Nightenom) [inline src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java:31]: components literal would be better

File: src/main/resources/assets/minecolonies/gui/townhall/windowcantfoundcolony.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        xsi:noNamespaceSchemaLocation="file:../../../../../api/java/com/minecolonies/blockui/blockui.xsd">
+    <image source="minecolonies:textures/gui/builderhut/builder_paper_short.png" size="190 130"/>
+
+    <buttonimage id="cancel" size="14 15" pos="165 9" source="minecolonies:textures/gui/button_x.png" textcolor="black"/>
+
+    <text id="text1" size="160 90" color="black" pos="15 15" textscale="1"/>
+
+    <buttonimage id="pickup" size="129 17" pos="30 100" textcolor="black"
+                 source="minecolonies:textures/gui/builderhut/builder_button_medium_large.png"
+                 disabled="minecolonies:textures/gui/builderhut/builder_button_medium_large_disabled.png" disabledsize="129 17"
```
Comment (Nightenom) [inline src/main/resources/assets/minecolonies/gui/townhall/windowcantfoundcolony.xml]: disabled size seems unnecessary (also in other xmls)

**Review by Nightenom — DISMISSED** [review]

File: src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java:22]: no

File: src/main/java/com/minecolonies/core/client/gui/townhall/WindowTownHallDeleteAbandonColony.java (hunk 2 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/core/client/gui/townhall/WindowTownHallDeleteAbandonColony.java:83]: might seem like it, but depending on the case I enable/disable the abandoned one separately anyway

Comment (Nightenom) [reply src/main/java/com/minecolonies/core/client/gui/townhall/WindowTownHallDeleteAbandonColony.java:83]: honestly, could be one cancel button but whatever

File: src/main/java/com/minecolonies/core/client/gui/WindowSupplyStory.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        else
+        {
+            story.add(Component.translatable("com.minecolonies.core.gui.supplyship.guide"));
+        }
+
+        this.findPaneOfTypeByID("text", Text.class).setText(story);
+        this.findPaneOfTypeByID("place", Button.class).setText(Component.translatable("com.minecolonies.core.gui.supplies.place", Component.translatable(stack.getItem().getDescriptionId())));
+    }
+
+    private void place()
```
Comment (someaddons) [inline src/main/java/com/minecolonies/core/client/gui/WindowSupplyStory.java:105]: no javadoc? which century is it?!

File: src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import net.minecraft.server.packs.resources.SimpleJsonResourceReloadListener;
+import net.minecraft.util.profiling.ProfilerFiller;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Loads and listens to custom visitor data added
```
Comment (someaddons) [inline src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java]: obligatory copy&paste error

File: src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+ */
+public class ColonyStoryDataListener extends SimpleJsonResourceReloadListener
+{
+    /**
+     * Gson instance
+     */
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+
+    /**
+     * List of custom visitor data
```
Comment (someaddons) [inline src/main/java/com/minecolonies/core/datalistener/ColonyStoryDataListener.java]: obligatory copy&paste error#2

**Review by someaddons — DISMISSED** [review]

**Review by someaddons — APPROVED** [review]

**Pull request merged**